### PR TITLE
ScrollView - implement onScroll() event

### PR DESCRIPTION
**Pull request by EugeniyKiyashko**

### CODE REVIEW

File: js/renovation/ui/scroll_view.tsx [hunk 1]
```diff
@@ -28,9 +32,18 @@ export interface ScrollOffset {
...
+  reachedBottom: boolean;
+  reachedLeft: boolean;
+  reachedRight: boolean;
+  reachedTop: boolean;
+}
+
 export type ScrollViewDirection = 'both' | 'horizontal' | 'vertical';
 
-export const ensureLocation = (location: number | Partial<Location>): Location => {
+export const ensureLocation = (
```
Comment (MikeVitik) [inline js/renovation/ui/scroll_view.tsx:51]: Why do you export it? I find import ensureLocation in a test, but it doesn't use in tests.

File: js/renovation/ui/scroll_view.tsx [hunk 2]
```diff
@@ -79,16 +92,26 @@ export const viewFunction = ({
 /* istanbul ignore next: class has only props default */
 @ComponentBindings()
 export class ScrollViewProps {
-  @Slot() children?: any;
+  @Slot() children?: JSX.Element | (JSX.Element | undefined | false | null)[];
 
   @OneWay() direction: ScrollViewDirection = 'vertical';
+
+  @Event() onScroll: (e: {
```
Comment (MikeVitik) [inline js/renovation/ui/scroll_view.tsx]: Use EventCallback<>

Comment (MikeVitik) [reply js/renovation/ui/scroll_view.tsx]: And no use noop function as default value.

File: js/renovation/ui/scroll_view.tsx [hunk 3]
```diff
@@ -99,20 +122,21 @@ export default class ScrollView extends JSXComponent<ScrollViewPropsType>() {
   }
 
   @Method()
-  scrollBy(distance: number | Partial<Location>): void {
-    const { direction } = this.props;
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
```
Comment (MikeVitik) [inline js/renovation/ui/scroll_view.tsx]: Why do you use any but not ensureLocation parameter type.

**Review by MikeVitik — CHANGES_REQUESTED** [review]

File: js/renovation/ui/scroll_view.tsx (hunk 1 above)
Comment (EugeniyKiyashko) [reply js/renovation/ui/scroll_view.tsx:51]: We use it there - https://github.com/DevExpress/DevExtreme/blob/c8881d706f11d697bc9ba1d2c6ce789a25c18b92/js/renovation/ui/__tests__/scroll_view.test.tsx#L711

File: js/renovation/ui/scroll_view.tsx (hunk 3 above)
Comment (EugeniyKiyashko) [reply js/renovation/ui/scroll_view.tsx]: We have a restriction in our generators - https://github.com/DevExpress/devextreme-renovation/issues/519

File: js/renovation/ui/scroll_view.tsx (hunk 2 above)
Comment (EugeniyKiyashko) [reply js/renovation/ui/scroll_view.tsx]: Fix https://github.com/DevExpress/DevExtreme/pull/14983/commits/598a71c95d324af86ad4169a478dce60f80eb409

**Review by MikeVitik — APPROVED** [review]

**Review by pavelgruba — APPROVED** [review]

**Review by jsdmitry — APPROVED** [review]

**Pull request merged**